### PR TITLE
Replace JSPM with Webpack

### PR DESCRIPTION
**Pull request by mbarton**
The build for Restorer requires JSPM and by extension a GitHub access token, which was running attached to my GitHub user. I got an email saying that it was using an [old authentication mechanism](https://developer.github.com/changes/2020-02-10-deprecating-auth-through-query-param) and JSPM itself was [just shut down](https://jspm.org/jspm-dev-release#sunsetting-the-cli).

Sooo I decided to migrate the project to Webpack as that is what most of the tools use these days. I also removed some old code in `app/assets` which I don't think is used any more (maybe it was Restorer v1?)

I tried to update Angular but got mysterious [errors relating to ng-transclude](https://github.com/acjim/angular-html-window/issues/3) so have kept it on the previous version.

### CODE REVIEW

File: app/views/main.scala.html [hunk 1]
```diff
@@ -5,17 +5,14 @@
 <head lang="en">
     <meta charset="UTF-8">
     <title>Composer Restorer - @pageTitle</title>
-    <link rel="stylesheet" type="text/css" href="@routes.Assets.at("jspm_packages/github/guardian/scribe-plugin-noting@0.3.29/skins/gu-noting.css")" />
```
Comment (mbarton) [inline app/views/main.scala.html:8]: This package depends on bower so won't even install in a fresh NPM project 🤦 

I've copy pasted the CSS into this project instead. Presumably this hasn't kept pace with print only/web only and other ProseMirror powered fun so is it a big deal?

File: public/javascripts/app/collections/SnapshotIdModels.js [hunk 2]
```diff
@@ -16,7 +14,7 @@ SnapshotIdModelsMod.factory('SnapshotIdModels', [
         class SnapshotIds extends BaseCollection {
             constructor(models){
                 super();
-                this.models = models.map((snapshot) => new SnapshotIdModel.getModel(snapshot)).sort(this.comparator);
+                this.models = models.map((snapshot) => SnapshotIdModel.getModel(snapshot)).sort(this.comparator);
```
Comment (mbarton) [inline public/javascripts/app/collections/SnapshotIdModels.js:17]: `getModel` calls `new` inside so you get an error. Presumably whatever ancient version of Babel or something we had was totally fine with this before

File: public/javascripts/app/lib/icons/index.js [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
-import publishActive    from './svg/publish-active.svg!text';
-import publishDisabled  from './svg/publish-disabled.svg!text';
-import previewActive    from './svg/preview-active.svg!text';
-import arrowDown        from './svg/arrow-down.svg!text';
-import composerIcon     from './svg/composer-icon.svg!text';
-import wrenchActive     from './svg/wrench-active.svg!text';
-import wrenchDisabled   from './svg/wrench-disabled.svg!text';
-import expandActive     from './svg/expand-active.svg!text';
-import expandDisabled   from './svg/expand-disabled.svg!text';
+import clockActive      from '!!raw-loader!./svg/clock-active.svg';
```
Comment (mbarton) [inline public/javascripts/app/lib/icons/index.js:3]: The funny syntax is to drop the existing SCSS loader and as such bring in the file without conversion to a data URL

File: public/javascripts/app/services/AnalyticsService.js [hunk 4]
```diff
@@ -1,6 +1,5 @@
 import angular from 'angular';
 import mediator from '../utils/mediator';
-import md5 from 'md5';
```
Comment (mbarton) [inline public/javascripts/app/services/AnalyticsService.js:3]: Not used

**Comment by mbarton** [comment]
I don't know why it's showing a Snyk failure. When I click through, all the failures are related to JSPM which I have removed in this PR?

**Comment by mbarton** [comment]
The [Teamcity job passes](https://teamcity.gutools.co.uk/viewType.html?buildTypeId=EditorialTools_FlexibleRestorer&branch_EditorialTools=mbarton%2Fremove-jspm&tab=buildTypeStatusDiv) but has the node version hard-coded in the build step so I had to re-configure the build first.

Not sure why it's not reporting build status on this PR. Worth a look once we're happy with the code changes

**Comment by jennygrahamjones** [comment]
When I first ran `sbt`, I got the following:
```
~/code/flexible-restorer  mbarton/remove-jspm ✔ 
▶ sbt
jenv: version `1.7' is not installed
```
I tried changing my local `.java-version` file to use 1.8 instead of 1.7. The tool still seems to run okay and tests pass, so it seems safe to bump up the required Java version. Alternatively, we could add something to the README about Java 1.7 being a requirement.

**Comment by mbarton** [comment]
Ah great spot, have updated it to 1.8

**Pull request merged**